src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/Marshal/Copy/ByteArrayTests.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class ByteArrayTests
+    {
+        public static readonly object[][] ArrayData =
+        {
+            new object[] { new byte[]{ 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 } }
+        };
+
+        [Theory]
+        [MemberData(nameof(ArrayData))]
+        public void NullValueArguments_ThrowsArgumentNullException(byte[] TestArray)
+        {
+            byte[] array = null;
+            Assert.Throws<ArgumentNullException>("destination", () => Marshal.Copy(TestArray, 0, IntPtr.Zero, 0));
+            Assert.Throws<ArgumentNullException>("source", () => Marshal.Copy(array, 0, new IntPtr(1), 0));
+            Assert.Throws<ArgumentNullException>("destination", () => Marshal.Copy(new IntPtr(1), array, 0, 0));
+            Assert.Throws<ArgumentNullException>("source", () => Marshal.Copy(IntPtr.Zero, TestArray, 0, 0));
```

Comment:
We use AssertExtensions.Throws instead of Assert.Throws when validating argument exception parameter names, because it takes into account implementations like UWP release builds where metadata is thrown away.

Reply:
Fixed, see in next commit